File: frontend/src/app/core/services/consumer.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { AddConsumerRequest, ConnectionStatusUpdateRequest, Consumer } from '../models/consumer.model';
import { PageResponse } from '../models/page-response.model';

@Injectable({ providedIn: 'root' })
export class ConsumerService {
  private readonly baseUrl = '/api/consumers';

  constructor(private readonly http: HttpClient) {}

  add(request: AddConsumerRequest): Observable<Consumer> {
    return this.http.post<Consumer>(this.baseUrl, request);
  }

  list(section: string | null, type: string | null, page: number, size: number): Observable<PageResponse<Consumer>> {
    let params = new HttpParams().set('page', page).set('size', size);
    if (section) params = params.set('electricalSection', section);
    if (type) params = params.set('customerType', type);
    return this.http.get<PageResponse<Consumer>>(this.baseUrl, { params });
  }

  myConsumers(): Observable<Consumer[]> {
    return this.http.get<Consumer[]>(`${this.baseUrl}/me`);
  }

  updateConnectionStatus(consumerNumber: string, request: ConnectionStatusUpdateRequest): Observable<Consumer> {
    return this.http.put<Consumer>(`${this.baseUrl}/${consumerNumber}/connection-status`, request);
  }
}

File: backend/src/main/java/com/electricity/billing/service/ConsumerService.java
package com.electricity.billing.service;

import com.electricity.billing.dto.request.AddConsumerRequest;
import com.electricity.billing.dto.request.ConnectionStatusUpdateRequest;
import com.electricity.billing.dto.response.ConsumerResponse;
import com.electricity.billing.dto.response.PageResponse;
import com.electricity.billing.entity.enums.CustomerType;
import com.electricity.billing.entity.enums.ElectricalSection;

import java.util.List;

public interface ConsumerService {

    ConsumerResponse addConsumer(AddConsumerRequest request);

    PageResponse<ConsumerResponse> listConsumers(ElectricalSection section, CustomerType type, int page, int size);

    ConsumerResponse updateConnectionStatus(String consumerNumber, ConnectionStatusUpdateRequest request);

    List<ConsumerResponse> myConsumers();
}
